kahuna/public/js/search/query.js

```diff
@@ -144,7 +144,9 @@ query.controller('SearchQueryCtrl', [
 
 
     $scope.$watchCollection(() => ctrl.filter, onValChange(filter => {
+        $rootScope.nonFree = ctrl.filter.nonFree ? ctrl.filter.nonFree : false;
```

Comment:
I can't see these properties being used on `$rootScope` elsewhere. `$rootScope` is effectively a global namespace – is there a reason we wouldn't scope these properties as variables defined within the controller? E.g.

```js
let isNonFree = false;

$scope.$watchCollection(() => ctrl.filter, onValChange(filter => {
    isNonFree = ctrl.filter.nonFree ? ctrl.filter.nonFree : false;
    // ... etc.
```

Reply:
Thanks, @jonathonherbert for the review,
using the $rootScope to keep the nonFree value across pages,
using the controller variable will not keep the nonFree filter value when moving to another page and the Free-to-use checkbox will not reflect the current state

E.g:
When a "Free to Use Only" tick has been selected to view,
and then click on an image to view
Then Return to Search
The tick  disappears however The filter is still being applied

Reply:
Gotcha, the requirement makes sense.

Can I recommend writing a little singleton service to keep track of this state, rather than adding to `$rootScope`? It'll more clearly encapsulate the values and what they're for, and will avoid adding things to a global variable. (`$rootScope` will become crowded and confusing quite quickly if we continue to take this approach elsewhere.)

I'm thinking really small – an example in this codebase might be `services/api/edits-api.js`, which uses some local state. Happy to find another example or explain further if that's not clear.

I'd be super happy to pair.

Reply:
Resolved in favour of using `localStorage` in https://github.com/guardian/grid/pull/3645/commits/d5bc5006460f3271a84f840328c1af5c0299ac7f.